maven-extension/README.md

```diff
@@ -57,42 +57,88 @@ mvn verify
+| `maven.repository.username` | Username used to upload artifacts to the Maven repository. See [Maven POM reference / Repository](https://maven.apache.org/pom.html#repository)     |
+| `peer.service`              | Maven repository hostname deduced from the Repository URL                                                                                           |
+| `http.url`                  | Base URL of the uploaded artifact `${repository.url}/${groupId}/${artifactId}/${version}` where the `.` of `${groupId}` are replaced by `/`         |
+| `http.method`               | `POST`                                                                                                                                              |
+
+The `span.kind` is set to `client`
+
+### `spring-boot:build-image`
+
+| Span attribute                | Description                                                                                                                                                                              |
+|-------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `container.image.name`        | Docker image specified in the `build-image` goal configuration                                                                                                                           |
+| `container.image.tag`         | Docker tag specified in the `build-image` goal configuration                                                                                                                             |
+| `container.registry.url`      | URL of the container registry to which this image is uploaded.  Attribute only added when the `build-image` goal publishes the Docker image.                                             |
+| `container.registry.username` | Username used to upload the image to the container registry.  Attribute only added when the `build-image` goal publishes the Docker image.                                               |
+| `peer.service`                | Docker Registry hostname. Attribute only added when the `build-image` goal publishes the Docker image.                                                                                   |
+| `http.url`                    | URL on the Docker registry deduced from the Docker image specified in the `build-image` goal configuration. Attribute only added when the `build-image` goal publishes the Docker image. |
+| `http.method`                 | `POST`. Attribute only added when the `build-image` goal publishes the Docker image.                                                                                                     |
+
+The `span.kind` is set to `client`
+
+TODO How can we report cases when multiple Docker image tags are published?  Introducing `container.image.tags`?
```

Comment:
Looking at the semantic conventions, the `container.image` attributes are specific to a compute unit where an image is running. As we're dealing with creating images, they won't fit very well.

Maybe we need a `container.build.image.name`, to indicate it's a build vs a runtime? I'm also not sure how free we are to experiment with new attributes. @trask do you have any thoughts?

Reply:
Thanks @kenfinnigan! We enter an exciting conversation. I know that the Elastic Common Schema (ECS) philosophy is to try to reuse the same attribute names for the context and for the things produced. It's a difference we highlighted to the Otel Logs Sig when we presented ECS 18 months ago. It's the outcome of years and years of experience.

Following this philosophy of reuse, we would use `container.image.name` and `container.image.tag` when capturing docker image creations.
If we had to choose different attribute names, I see benefits in prefixing/suffixing the initial name rather than crafting a new name, I would prefer to look at `build.container.image.name` rather than `container.build.image.name`.

Reply:
I was wondering about the "right" place to put `build`, and I take your point that using it as a prefix provides the ability to convert any runtime attribute to a build captured one.

I'd be good with adding `build` in front, but don't know if we need to get that into the spec or we can experiment here first?

Reply:
I'd recommend including the instrumentation name in the attribute until it gets spec'd to avoid confusion, e.g. a couple of examples from other instrumentations:

* `spring-webflux.handler.type`
* `rabbitmq.command`

so in this case something like

`maven....`

> Looking at the semantic conventions, the `container.image` attributes are specific to a compute unit where an image is running. As we're dealing with creating images, they won't fit very well.

ya, I would avoid re-using existing attribute names in a way that doesn't match the spec currently.

I like the general idea of `build.*` attributes / semantic conventions. Whenever you're ready, I think it's worth starting this discussion in the spec repo and proposing something based on what you've found useful here.

Reply:
Note: as discussed with @v1v we will want to track "any" resource created by CI/CD/Deployment:
* binaries: jar file, executable file
* container images
* cloud resources produced by Ansible/Terraform/... like virtual machine, Kubernetes cluster, database instance, SNS topic...
* ...

Would we prefix by `build.*` ? or `produce.*` ?

Reply:
Based on what @trask said, I think we need to have a name like `maven.spring.container.image`, or something equally wordy but it needs to start with `maven.*`.

However, for proposing something to the spec, we can come up with a good root level name to represent build outputs. For this, I like `build.*` as it enables capturing of details specific to a build process, and not just what was produced by a build, but that might depend on how much down that road the spec group wants to go

Reply:
I've added handler for more Maven goals to better understand what attributes will look like.
We now handle `deploy:deploy`, `spring-boot:build-image`, `jib:build`, `snyk:test`, and `snyk:monitor` goals.
I still have to document this new goals in the README.md to clearly describe what will be the user experience and the attributes with this prefix `maven.build.*`